tutorials/graph-theory/prims-algorithm.md

```diff
@@ -1,12 +1,186 @@
+**Note:** The above link requires the Leetcode Subscription. To view the problem, use [this](https://leetcode.ca/all/1135.html) alternate link.
+
+### Instructions: 
+- There are N cities numbered from 1 to N.
+- You are given connections, where each `connections[i] = [city1, city2, cost]` represents the cost to connect `city1` and `city2 ` together. The connections are all bidirectional.
+- Return the minimum cost so that for every pair of cities, there exists a path of connections (possibly of length 1) that connects those two cities together.  The cost is the sum of the connection costs used. If the task is impossible, return $-1$.
+
+Example: 
+
+**Input:** N = `3`, connections = `[[1,2,5],[1,3,6],[2,3,1]]`
+
+**Output:** `6`
+
+### Approach
+First, we convert the given 'connections' array, into an adjacency list for ease of traversing.
+<Tabs>
+
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Infonioknight"/>
+  
+```py
+connections = [[1,2,5],[1,3,6],[2,3,1]] # Example
+adjacency_list = {}
+
+for connection in connections:
+    if connection[0] not in adjacency_list:
+        adjacency_list[connection[0]] = []
+    if connection[1] not in adjacency_list:
+        adjacency_list[connection[1]] = []
+    adjacency_list[connection[0]].append((connection[2], connection[1]))
+    adjacency_list[connection[1]].append((connection[2], connection[0]))
+```
+</TabItem>
+
+The adjacency list will look like:
+```
+1: [(5, 2), (6, 3)]
+2: [(5, 1), (1, 3)]
+3: [(6, 1), (1, 2)]
+```
+Where each tuple consists of '(cost, destination)'.
+
+Now, to find the minimum cost, we use the below approach.
+#### Explanation:
+
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Infonioknight"/>
+
+```py
+for connection in adjacency_list[start]:
+    heapq.heappush(connection_queue, connection)
+```
+</TabItem>
+
+First, we push all the connections of the starting node into the priority queue. On doing so, the nodes with edges of MINIMUM COST come in front.
+
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Infonioknight"/>
+  
+```py
+while len(seen) < n and connection_queue:
+    cost, current = heapq.heappop(connection_queue)
+    if current in seen:
+        continue
+
+    res += cost
+    seen.add(current)
+```
+</TabItem>
+
+- Then, we take the first element out of the queue (this is the node we ideally will add to our MST as it will have the minimum cost. We also ensure that the node selected hasn't already been visited.
+- If the node hasn't been visited, we then add the cost to our solution and mark the node as visited.
+
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Infonioknight"/>
+  
+```py
+    for connection in adjacency_list[current]:
+        if connection[1] not in seen:
+            heapq.heappush(connection_queue, connection)
+
+if len(seen) < n: return -1
+else: return res
```

Comment:
`else:` is not necessary here since you've returned something if the condition is true.